### PR TITLE
New documentation

### PR DESCRIPTION
**Pull request by LorenzoBaesso**
First part of the documentation. 

Major work/review needs to be done on the following folders:
- infrastructure
- lib/
- lib/crypto
- lib/packet

### CODE REVIEW

File: infrastructure/beacon_server.py [hunk 1]
```diff
@@ -100,6 +100,16 @@ class InterfaceState(object):
     IFID_TOUT = 3.5 * IFID_PKT_TOUT
 
     def __init__(self):
+        """
+        Constructor.
```
Comment (pszal) [inline infrastructure/beacon_server.py]: I'm not sure whether it helps to have these docstring without param names and where even number of params is incorrect (the same comment applies for almost all constructors below).
IMO it is better to have 10% of correct docstrings, than 100% of incorrect templates. 
What do you think?

Comment (LorenzoBaesso) [inline infrastructure/beacon_server.py]: Of course. Indeed it's not complete. But I need you to remove the classes that are not used anymore and after that I will pass again the docstrings.

File: ad_management/common.py [hunk 2]
```diff
@@ -1,3 +1,20 @@
+# Copyright 2014 ETH Zurich
```
Comment (pszal) [inline ad_management/common.py:1]: 2015 (the same for other new copyrights)

File: ad_management/secure_rpc_server.py [hunk 3]
```diff
@@ -1,10 +1,23 @@
 #!/usr/bin/env python3
```
Comment (pszal) [inline ad_management/secure_rpc_server.py:1]: is this our code?

Comment (tonyo) [inline ad_management/secure_rpc_server.py:1]: It's my experimental stuff, I'll add copyrights sometime soon.

Comment (tonyo) [inline ad_management/secure_rpc_server.py:1]: Oh well, I mean, yes, it's based on something from SO, but it's not a copy-paste.

Comment (pszal) [inline ad_management/secure_rpc_server.py:1]: ok!

**Pull request merged**